holidays/countries/yemen.py

```diff
@@ -0,0 +1,138 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import (
+    MAR,
+    APR,
+    MAY,
+    JUN,
+    JUL,
+    AUG,
+    SEP,
+    OCT,
+)
+from holidays.groups import InternationalHolidays, IslamicHolidays
+from holidays.holiday_base import HolidayBase
+
+
+class Yemen(HolidayBase, InternationalHolidays, IslamicHolidays):
+    """Yemen holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Yemen>
+        * <https://yemenembassy.it/festivities/>
+    """
+
+    country = "YE"
+    default_language = "ar"
+    # %s (estimated).
+    estimated_label = tr("%s (مُقدَّر)")
+    start_year = 1990
+    supported_languages = ("ar", "en_US")
```

Comment:
```suggestion
    supported_languages = ("ar", "en_US")
    weekend = {FRI}
```
According to the [Presidential Decree Law No. (2) of 2000](https://web.archive.org/web/20140714152707/http://www.presidentsaleh.gov.ye/showlaws.php?_lwbkno=3&_lwptno=2&_lwnmid=4):
> "If the holiday falls on a Friday or an official holiday, that day shall be replaced by another working day immediately following the holiday."

This meant we'll need to switch from `HolidayBase` to `ObservedHolidayBase` as well - I would recommend using [`saudi_arabia.py`](https://github.com/vacanza/holidays/blob/8b56d46159f85cf486ac50b9a1f049213b37c869/holidays/countries/saudi_arabia.py#L29) and [`indonesia.py`](https://github.com/vacanza/holidays/blob/8b56d46159f85cf486ac50b9a1f049213b37c869/holidays/countries/indonesia.py#L34) as a base here 👍

Reply:
> According to the [Presidential Decree Law No. (2) of 2000](https://web.archive.org/web/20140714152707/http://www.presidentsaleh.gov.ye/showlaws.php?_lwbkno=3&_lwptno=2&_lwnmid=4):
> 
> > "If the holiday falls on a Friday or an official holiday, that day shall be replaced by another working day immediately following the holiday."
> 
> This meant we'll need to switch from `HolidayBase` to `ObservedHolidayBase` as well - I would recommend using [`saudi_arabia.py`](https://github.com/vacanza/holidays/blob/8b56d46159f85cf486ac50b9a1f049213b37c869/holidays/countries/saudi_arabia.py#L29) and [`indonesia.py`](https://github.com/vacanza/holidays/blob/8b56d46159f85cf486ac50b9a1f049213b37c869/holidays/countries/indonesia.py#L34) as a base here 👍

Both Saudia and Indonesia have quite different approaches for ObservedHolidays. What to do?